meshnets/data_processing/from_hugging_face.py

```diff
@@ -0,0 +1,34 @@
+"""Function to process data from hugging face datasets to graphs.
+
+This module assumes the data is in the format of the hugging face
+datasets with the following columns:
+
+- nodes: list of nodes in the graph with shape [n, 3] where 3
+  corresponds to the spacial coordinates.
+
+- edges: list of edges in the graph with shape [m, 2] where 2
+  corresponds to the source and target nodes.
+
+- wind_vector: list with shapes (3,) that contains the wind speeds on
+  the 3 axis.
+
+- wind_pressure: list with shape (N,) that contains the wind pressure
+  on the nodes.
+
+"""
+import numpy as np
+
+
+def to_undirected(example):
+    """Makes the graph undirected.
+
+    This is equivalent to adding the inverse edges to the graph. That
+    is, if (i, j) is in the `edges` we also add the edge (j, i).
+
+    """
+    edges = np.array(example['edges'])
```

Comment:
there is another thing that is implicit here, which is a conversion from a list to a numpy array....

maybe you can decouple that? So your first map function is to_numpy and you convert all the list to numpy arrays.

Reply:
I understand your comment. Unfortunately that is not how hugging face works. The columns of the Hugging face datasets must be of a very specific set of features, e.g, Array2D, Array3D, text.... Numpy is not one of them, therefore numpy arrays are always represented as lists. As such the `to_numpy` function would be an irrelevant steps.

Take a look at the next few lines of code:

![image](https://github.com/inductiva/meshnets/assets/33664950/2d492c86-b2eb-47c6-9875-9ec31d0abd91)

Reply:
Hmm... Interesting....